extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -72,16 +78,35 @@ class AmpYoutube extends AMP.BaseElement {
     iframe.width = this.width_;
     iframe.height = this.height_;
     this.element.appendChild(iframe);
-    /** @private {?Element} */
+
+    /** @private {!Element} */
     this.iframe_ = iframe;
 
-    // TODO(mkhatib, #2050): Use PlayerReady message for layout promise.
-    return loadPromise(iframe);
+    /** @private @const {!Promise} */
+    this.playerReadyPromise_ = this.getPlayerReadyPromise_();
+
+    this.getWin().addEventListener(
```

Comment:
`iframe.ownerDocument.defaultView` is the same value as `this.getWin()`

The opposite of what you say is actually true: `this.getWin()` is guaranteed to get you the right window, while `window` might be a different thing.